src/DIRAC/Core/Tornado/Client/private/TornadoBaseClient.py

```diff
@@ -504,12 +530,38 @@ def _request(self, retry=0, outputFile=None, **kwargs):
     # getting certificate
     # Do we use the server certificate ?
     if self.kwargs[self.KW_USE_CERTIFICATES]:
-      cert = Locations.getHostCertificateAndKeyLocation()
+      auth = {'cert': Locations.getHostCertificateAndKeyLocation()}
+
+    # Use access token?
+    elif self.__useAccessToken:
+      # Read token from token environ variable or from token file
+      result = getLocalTokenDict()
+      if not result['OK']:
+        return result
+      token = result['Value']
+
+      # Check if access token expired
+      if token.is_expired():
+        if not token.get('refresh_token'):
+          return S_ERROR('Access token expired.')
+
+        # Try to refresh token
+        self.__idp.scope = None
+        result = self.__idp.refreshToken(token['refresh_token'])
+        if result['OK']:
+          token = result['Value']
+          result = writeTokenDictToTokenFile(token)
+        if not result['OK']:
+          return result
```

Comment:
```suggestion
        if not result['OK']:
          return result
        token = result['Value']
        result = writeTokenDictToTokenFile(token)
```

Reply:
Also you're not handling the error response from `writeTokenDictToTokenFile`

Reply:
here
```
if not result['OK']:
  return result
```
handle `S_ERROR` from
```
result = self.__idp.refreshToken(token['refresh_token'])
```
and 
```
result = writeTokenDictToTokenFile(token)
```

Sorry, I do not quite understand what is missing here?

Reply:
Ah sorry, I incorrectly parsed the code. I think it might be clearer to write it this way but it's up to you:

```suggestion
        if not result['OK']:
          return result
        token = result['Value']
        result = writeTokenDictToTokenFile(token)
        if not result['OK']:
          return result
```